### PR TITLE
Major-pin default plugins

### PR DESCRIPTION
**Pull request by yaauie**
Pins all default plugins _optimistically_ to the major.minor in the head of 8.17's `Gemfile.*.lock`, in order to avoid accidentally shipping _any_ plugin's breaking changes in the 8.x series.

This is a thing that @jsvd and I talked about, so I thought I'd put it up for discussion.

The major downside is that it _prevents_ a released artifact containing the resulting `Gemfile` from being _able_ to use a plugin that doesn't match, even if the user wants to. Notably, we have plans to release 8.x of `logstash-filter-elastic_integration` that is _not_ a breaking change from `1.x`, and something like this proposed change would affect that.

We _may_ need to have a way to specify dependencies for build-time resolution that are _not_ included in the artifact's `Gemfile`.

### CODE REVIEW

**Comment by jsvd** [comment]
> We may need to have a way to specify dependencies for build-time resolution that are not included in the artifact's Gemfile.

I see this only as a temporary solution until this branch is deleted.

**Review by donoghuc — APPROVED** [review]
Seems reasonable as an interim solution. While an end user modifying a Gemfile is probably not optimal and may have LS upgrade fragility it probably wouldnt be impossible if somehow they got in to the upgrade problem described.

File: Gemfile.template [hunk 1]
```diff
@@ -45,5 +44,99 @@ gem "thwait"
 gem "bigdecimal", "~> 3.1"
 
 # Pin plugins to avoid including those with breaking changes in development builds without a defined lockfile
-gem "logstash-input-beats", "~> 6"
-gem "logstash-output-http", "~> 5"
+gem "logstash-codec-avro", "~> 3.4", :require => false
```
Comment (mashhurs) [inline Gemfile.template:47]: When do we update these plugin versions?

File: build.gradle [hunk 2]
```diff
@@ -101,6 +101,7 @@ allprojects {
       "--add-opens=java.base/java.lang=ALL-UNNAMED",
       "--add-opens=java.base/java.util=ALL-UNNAMED"
     ]
+    maxHeapSize = "2g"
```
Comment (mashhurs) [inline build.gradle:104]: Hopefully we will not forget mentioning this in the release notes.

File: Gemfile.template (hunk 1 above)
Comment (yaauie) [reply Gemfile.template:47]: The optimistic `~>` pin means we don't need to. This keeps the `8.x` branch aligned with the major versions and at least the minor that we are shipping in 8.17.0, and should not be included in any released artifacts.

File: build.gradle (hunk 2 above)
Comment (yaauie) [reply build.gradle:104]: There would be no need; this is a build configuration that ensures the tests have sufficient memory to run and will not affect the distributed artifact in any way.

File: Gemfile.template (hunk 1 above)
Comment (mashhurs) [reply Gemfile.template:47]: Ah target branch is `8.x` 🤦!

**Review by mashhurs — APPROVED** [review]
LGTM as well.

**Comment by jsvd** [comment]
Just to clarify, this is in 8.x branch only.
Any release branch taken from 8.x like 8.18 or eventually 8.19 should not
use this extra list in its Gemfile.
We will use the standard Gemfile, take the lock file from the previous
minor, run the gh action to advance only minors like we have always done
and that’s it.
So we’ll never release Logstash with this Gemfile.

**Pull request merged**